app/models/payment_allocation_set.rb

```diff
@@ -11,7 +12,11 @@ def payment_percent_sum
 
   private
 
+  def filter_zero_percent_payment_allocations
+    self.payment_allocations = payment_allocations.select { |p| p.percent.nonzero? }
```

Comment:
Don't override the scope just return the scope.

Reply:
This is used in a before validation to filter out zero percents so I don't needlessly store 0% donations in the database. There a better place/way to handle this? filter outside the model?

Reply:
```suggestion
    payment_allocations.where('percentage != 0')
```